### ISSUE TITLE
Layout problems

### ISSUE THREAD

**Issue by oliverjam** [jackherizsmith/wardles#26]
You've got some funky stuff going on in your CSS that is causing layout issues. I'll quickly point out the rules causing most of the problems, then give some general guidance I find helpful for making layout easier.

## Broken stuff

You've got several elements that end up wider than a mobile viewport (about `320px-480px` or `20rem-30rem`). This causes the page to overflow horizontally, and then any elements set to fill the available width end up massive. The overflow isn't obvious because they fill the empty space.

Your nav component stays in a flex row without wrapping, which never fits on a mobile viewport. There are too many things trying to fit in a row, so you'll probably have to rethink the design. I'm not sure the float is helping either.

One of the sections has percentage padding that is also causing overflow:

https://github.com/jackherizsmith/wardles/blob/95aec7b083fa21ca330404e4503787390e8b5df6/src/sass/pages/_home.scss#L95

Your article component has a hard-coded width that is wider than most mobile viewports:

https://github.com/jackherizsmith/wardles/blob/95aec7b083fa21ca330404e4503787390e8b5df6/src/sass/components/_article-card.scss#L3

Getting rid of these 3 things made the layout normal for me on a mobile viewport.

Also the big photo at the top of the page is really tall, not sure if that's deliberate. It's ~`185vh` on mobile, so you have to scroll for a while to get past it.

**Comment by oliverjam** [comment]
## General advice

The main takeaway should be "work with the browser". Fighting against it by trying to set loads of stuff and control everything is exhausting.

I highly recommend this resource if you want to learn how to write easier and more resilient layout CSS: https://every-layout.dev/

Some people are calling this style "Intrinsic Layout" or "Resilient Design". [Jen Simmons](https://talks.jensimmons.com) has been popularising it—her YouTube channel [Layout Land](https://www.youtube.com/channel/UC7TizprGknbDalbHplROtag) has some great stuff. Here's a [fun intro](https://adactio.com/journal/14889) to the ideas as well.

### Never set widths

The browser lays out block elements as 100% width by default. Try and let everything fill its containing element, and _constrain_ widths using `max-width`.

The second you set an absolute `width` value you're asking for things to not fit in their container and overflow.

### Avoid percentages

Percentages are hard to work with as they are relative. It's usually better to rely on `max-width` and absolute `padding` values to constrain an elements size. Percentages are weird because e.g. `width: 60%` keeps shrinking forever.

### Never use floats

Floats were a layout hack from the before-time when we didn't have flex/grid. If you need to break out of the standard layout flow (blocks stacking on top of each other) then always use flex or grid.

### Avoid absolute positioning

Absolutely positioned elements are taken out of the normal browser layout flow. You are now entirely responsible for ensuring they're in the right place and at the right size. It's so easy to end up with overflowing absolutely positioned elements.

You can replace most uses of absolute positioning with CSS grid nowadays.

### Write resilient layouts

When you use flexbox/grid to put things in a row think about what happens when there isn't enough space. What if you add more items to the row? What if the container gets rendered inside another narrower element?

It's usually a good idea to let flex containers wrap (and to use auto grid columns) so your content adapts to changing dimensions.

**Comment by oliverjam** [comment]
Oh also I get why people like the `font-size: 62.5%` trick, but it can impact accessibility (preventing users controlling the default font-size). It's probably better to just get used to thinking and working in `rem`.

From a design perspective it's better to work with a scale anyway, and I find thinking in pixels encourages you to write weird specific in-between values. I pretty much just go with:
`0.875rem, 1rem, 1.25rem, 1.5rem, 2rem, 4rem` etc and don't worry about what the corresponding pixel values are.

**Comment by jackherizsmith** [comment]
Hey sorry this took me so long to respond to. Thank you so, so much for your detailed and thoughtful input. 

The 62.5% thing was in an Udemy course I did and was actually going to ask you about this, since when he introduced the idea he said being a percentage ensured it remained accessible and I wasn't too sure about it. I will go back to just using good old `rem`.

Everything else here is so helpful and insightful and a lot to go at whilst I rewrite my styling. At the core, I failed to build mobile first, so when I added the viewport meta tag (way too late) it was bound to be broken - I won't be making that mistake again!

I hadn't appreciated that about `width`s and percentages before and that totally makes sense. I'll go back through asap to rectify these mistakes.

Thanks again so much for looking through my code!
